modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java

```diff
@@ -1216,6 +1216,10 @@ else if (Objects.equals(clientExtension.type, "siteInitializer")) {
 				clientExtension, "membershipType", "open", "private",
 				"restricted");
 		}
+		else if (Objects.equals(clientExtension.type, "themeCSS")) {
+			_validateTypeSettingsValues(
+				clientExtension, "scope", "controlPanel", "page");
+		}
```

Comment:
What do you think about create constants for these Strings?

Reply:
This is good. I am planning on refactoring the way we declare fields and their valid values for frontend client extensions. I will do it for all of them and send a PR after finishing this one and the admin GlobalCSS cx. Is it okay for you if I do it later?

Reply:
My idea is to make it simple to declare valid values for each attribute of a client extension type. Maybe the usage could be

```java
else if (Objects.equals(clientExtension.type, "themeCSS")) {
	_validateTypeSettingsValues(
		clientExtension, CETConstants.ThemeCSS.Scope, CETConstants.ThemeCSS.Scope.values());
}
```

Reply:
Actually, never mind, I will do it for this type of CX and use it as an example to propose a global update for the other types.

Reply:
After some talking with Anderson, we will be planning a refactor to make it easy to read and write code for frontend CX. The idea is to make it simple to declare fields and values. 

We will be contacting the frontend and dev-tools teams

Reply:
Yeah improving this would be good.